internal/pipeline/merging.go

```diff
@@ -302,10 +309,17 @@ func (m *filesystemMerging) WithEachMerged(f func(int, upload.Agent, *ach.File)
 		} else {
 			logger.Info().Log("we are the leader")
 
-			if err := f(i, agent, files[i]); err != nil {
+			filename, err := f(i, agent, files[i])
+			if err != nil {
 				el.Add(fmt.Errorf("problem from callback: %v", err))
 			} else {
 				successfulRemoteWrites++
+
+				processedFiles.Merges = append(processedFiles.Merges, newMergedFile(
+					filename,
+					m.shard.Name,
+					matches, // TODO(adam): shouldn't include all matches, but how to tell?
```

Comment:
This is a bit complicated... `matches` is an `[]string` of filenames that we're support to filter against `file[i]`, but there's no good way to compare a filename to the `*ach.File` without reading every file again.